soc/arm64/intel_socfpga/agilex/mmu_regions.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2021, Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arch/cpu.h>
+#include <arch/arm64/arm_mmu.h>
+
+static const struct arm_mmu_region mmu_regions[] = {
+
+	/* System manager register that required by clock driver */
+	MMU_REGION_FLAT_ENTRY("SYSTEM_MANAGER",
+			      0xFFD12000,
+			      0x00001000,
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_DEFAULT_SECURE_STATE),
```

Comment:
syscon-like device? I am not sure how syscon-like device is, so i can't confirm.

Agilex System Manager register:
https://www.intel.com/content/www/us/en/programmable/hps/agilex/hps.html#jwd1561687644789.html

System Manager is a block inside HPS (Hard processer system) in Agilex SoCFPGA, it provide some settings for the periperal interface within HPS. 

Clock driver will need to read the external clock freq from boot_scratch_cold1 and boot_scratch_cold2 registers in system manager clock. cold1 and cold2 register is defined for custom used, in our implementation, we  use these 2 register to stores the external system clock system from user settings. So we need to access it in clock driver to calculate the frequently.

Reply:
**syscon** comes from the linux world https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/mfd/syscon.yaml and it represents  _a register region containing a set of miscellaneous registers. The registers are not cohesive enough to represent as any specific type of device_ and unfortunately this is exactly what it looks like to me.

Now, the problem is that you are defining here a device region without a backing device or driver and that should be always avoided but on the other hand Zephyr is not shipping any syscon driver for that.

I cannot ask you to write a syscon driver (even though it would be really easy to do) but please, at least use a couple of defines for the reg / size values and add a comment explaining what this region is.

Reply:
I have taka a look at syscon.yaml, and, yes that this is what I need. The system manager register will be access by different driver or feature in future, we just need a way to access the register directly.

I agree with you that it is cleaner to create a "syscon" device tree node for these register range. I can add this new device tree node element and driver. If there any concern if I proceed to add it?

Can we use back "syscon" as the name for this element? I expect this element only need to support 2 properties: "compatible" and "reg". We can add more properties in future when needed.

Reply:
I will proceed to add a driver call system_mgr for our device.

Reply:
I'm writing a syscon driver right now, I will push it tomorrow.

Reply:
Noted, thanks. I will wait for your change. :)